common/database.cpp

```diff
@@ -1128,979 +1042,850 @@ bool Database::CheckNameFilter(std::string name, bool surname)
-	auto query = fmt::format(
-		"SELECT `id` FROM `character_data` WHERE `name` = '{}'",
-		name
-	);
-	auto results = QueryDatabase(query);
-	if (!results.Success() || results.RowCount()) {
-		return false;
-	}
+	e.name = new_name;
 
-	return true;
+	return CharacterDataRepository::UpdateOne(*this, e);
 }
 
-uint8 Database::GetServerType() {
-	std::string query("SELECT `value` FROM `variables` WHERE `varname` = 'ServerType' LIMIT 1");
-	auto results = QueryDatabase(query);
-	if (!results.Success()) {
+bool Database::CheckUsedName(const std::string& name)
+{
+	return !CharacterDataRepository::GetWhere(
+		*this,
+		fmt::format(
+			"`name` = '{}'",
+			Strings::Escape(name)
+		)
+	).empty();
+}
+
+uint32 Database::GetServerType()
+{
+	const auto& l = VariablesRepository::GetWhere(*this, "`varname` = 'ServerType' LIMIT 1");
+
+	if (l.empty()) {
 		return 0;
 	}
 
-	if (results.RowCount() != 1)
-		return 0;
+	auto e = l.front();
 
-	auto row = results.begin();
-	return Strings::ToUnsignedInt(row[0]);
+	return Strings::ToUnsignedInt(e.value);
 }
 
 bool Database::MoveCharacterToZone(uint32 character_id, uint32 zone_id)
 {
-	std::string query = StringFormat(
-		"UPDATE `character_data` SET `zone_id` = %i, `x` = -1, `y` = -1, `z` = -1 WHERE `id` = %i",
-		zone_id,
-		character_id
-	);
+	auto e = CharacterDataRepository::FindOne(*this, character_id);
 
-	auto results = QueryDatabase(query);
+	e.zone_id = zone_id;
 
-	if (!results.Success()) {
-		return false;
-	}
+	e.x = e.y = e.z = -1;
 
-	return results.RowsAffected() != 0;
+	return CharacterDataRepository::UpdateOne(*this, e);
 }
 
-bool Database::MoveCharacterToZone(const char *charname, uint32 zone_id)
+bool Database::MoveCharacterToZone(const std::string& name, uint32 zone_id)
 {
-	std::string query = StringFormat(
-		"UPDATE `character_data` SET `zone_id` = %i, `x` = -1, `y` = -1, `z` = -1 WHERE `name` = '%s'",
-		zone_id,
-		charname
+	auto l = CharacterDataRepository::GetWhere(
+		*this,
+		fmt::format(
+			"`name` = '{}'",
+			Strings::Escape(name)
+		)
 	);
 
-	auto results = QueryDatabase(query);
-
-	if (!results.Success()) {
+	if (l.empty()) {
 		return false;
 	}
 
-	return results.RowsAffected() != 0;
-}
+	auto e = l.front();
 
-uint8 Database::GetRaceSkill(uint8 skillid, uint8 in_race)
-{
-	uint16 race_cap = 0;
+	e.zone_id = zone_id;
 
-	//Check for a racial cap!
-	std::string query = StringFormat("SELECT skillcap from race_skillcaps where skill = %i && race = %i", skillid, in_race);
-	auto results = QueryDatabase(query);
+	e.x = e.y = e.z = -1;
 
-	if (!results.Success())
-		return 0;
-
-	if (results.RowCount() == 0)
-		return 0;
-
-	auto row = results.begin();
-	return Strings::ToUnsignedInt(row[0]);
+	return CharacterDataRepository::UpdateOne(*this, e);
 }
 
-uint8 Database::GetSkillCap(uint8 skillid, uint8 in_race, uint8 in_class, uint16 in_level)
+uint32 Database::GetCharacterInfo(
+	const std::string& name,
+	uint32* account_id,
+	uint32* zone_id,
+	uint32* instance_id
+)
```

Comment:
This function is used in one place and should be removed and repository used inline where it is in the code. These pointer return loader functions are gross